### PR TITLE
Release 2.0.0-rc.1

### PR DESCRIPTION
**Pull request by WanzenBug**
Prepare the first release candidate of operator v2 for release.

Mostly replaces the referenced images and cleans up some docs.

### CODE REVIEW

**Comment by JoelColledge** [comment]
> @JoelColledge I intend to switch the default branch to `v2`, so people will start seeing it more prominently, even in the current "preview" state. Any thoughts?

Sounds like a good plan to me. The community Piraeus operator here on GitHub can be switched to default to `v2`.

Once there is a stable release of `v2`, I think we should "merge" `v2` into `master` and branch off a `v1` maintenance branch.

**Comment by JoelColledge** [comment]
Do we want to prepare this release with multiple commits like other releases? https://github.com/piraeusdatastore/piraeus-operator/pull/337/commits

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [v2.0.0-rc.1] - 2022-02-21
+
+### Breaking
+
+- Removed existing CRD `LinstorController`, `LinstorSatelliteSet` and `LinstorCSIDriver`.
+- Helm chart deprecated in favor of new `kustomize` deployment.
+- Helm chart changed to only deploy the Operator, not the operands.
```
[the remark this answers is not in the document]
Comment (WanzenBug) [reply CHANGELOG.md:16]: :smile: I think this is "operator-sdk speak". I better change it to something understandable :thinking: 

> ... deploy the Operator. The `LinstorCluster` resource to create the storage cluster needs to be created separately.

?

**Comment by WanzenBug** [comment]
> Do we want to prepare this release with multiple commits like other releases?

I was thinking about that, but I don't really see the advantage any more. I don't quite remember why we did it that way in the first place.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoelColledge) [reply CHANGELOG.md:16]: `operands` is used in academic mathematics/informatics contexts, so it isn't that unreasonable.

Your suggestion is good.

**Comment by JoelColledge** [comment]
> I was thinking about that, but I don't really see the advantage any more. I don't quite remember why we did it that way in the first place.

You introduced it, I believe :smile: It think the main advantage is that when someone deploys from `master`/`v2` they should get the `latest` image rather than the most recent release.

**Comment by WanzenBug** [comment]
Yeah, I remember introducing it. I guess there is a minor advantage with having a "rolling" release option. Will work on it tomorrow.

**Comment by WanzenBug** [comment]
Added it now. On the intended "v2.0.0-rc.1" commit, the docs will point to the exact version, during normal development it will just use the `v2` reference. That means that deploying `?ref=v2.0.0-rc.1` will pull in the exact image version, too.

I didn't bother with resetting the helm chart, as it will only be updated sporadically in any case.

**Review by JoelColledge — APPROVED** [review]

**Pull request merged**